### PR TITLE
Imagestream and Buildconfigs unit tests

### PR DESCRIPTION
**Pull request by ashetty1**
Adds test for `NewAppS2I`, `GetImageStreams` func in `pkg/occlient/occlient.go`

### CODE REVIEW

**Comment by ashetty1** [comment]
Up for the first review @cdrage @kadel @syamgk

**Comment by syamgk** [comment]
LGTM

File: pkg/occlient/occlient_test.go [hunk 1]
```diff
@@ -365,3 +453,462 @@ func Test_parseImageName(t *testing.T) {
...
+	labels := cmplabels.GetLabels("ruby", "apptmp", true)
+	labels[cmplabels.ComponentTypeLabel] = "ruby"
+
+	tests := []struct {
+		name      string
+		namespace string
+		args      args
+		wantErr   bool
+	}{
+		{
```
[the remark this answers is not in the document]
Comment (ashetty1) [reply pkg/occlient/occlient_test.go]: Hmm, it is getting imagestreams. Any suggestion on what the second test could be?

[the remark this answers is not in the document]
Comment (kadel) [reply pkg/occlient/occlient_test.go]: - tests should also verify that error is returned when it should be returned 
    - builderImage that doesn't exist
    - empty builderImage

- verify that if empty `gitURL` is passed that BuildConfig has BuildSourceBinary 
- verify that if `gitURL` is not empty than BuildConfig has GitBuildSource
...
...

File: pkg/occlient/occlient_test.go [hunk 2]
```diff
@@ -529,3 +615,576 @@ func Test_parseImageName(t *testing.T) {
...
+			if len(fkclientset.AppsClientset.Actions()) != 1 {
+				t.Errorf("expected 1 AppsClientset.Actions() in NewAppS2I, go: %v", fkclientset.AppsClientset.Actions())
+			}
+
+			if len(fkclientset.Kubernetes.Actions()) != 1 {
+				t.Errorf("expected 1 Kubernetes.Actions() in NewAppS2I, go: %v", fkclientset.Kubernetes.Actions())
+			}
+
+			if err == nil {
+				createdIS := fkclientset.ImageClientset.Actions()[2].(ktesting.CreateAction).GetObject().(*imagev1.ImageStream)
```
Comment (kadel) [inline pkg/occlient/occlient_test.go]: why are we checking just `3rd` action?  What if the order in which ImageStream is created changes?

**Comment by ashetty1** [comment]
The tests are failing because our functions don't have checks for empty strings. @kadel, kindly check if this looks okay

File: pkg/occlient/occlient_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (kadel) [reply pkg/occlient/occlient_test.go]: One important test that I think we are still missing is non-empty builderImage that doesn't' exist.

[the remark this answers is not in the document]
Comment (ashetty1) [reply pkg/occlient/occlient_test.go]: @kadel how to do that?  You mean not have a valid response through reactor?

**Comment by ashetty1** [comment]
@cdrage those failing tests have been commented now.

**Pull request merged**